homework_3week.js

```diff
@@ -0,0 +1,46 @@
+/* Написать свою реализацию методов map, filter которые будут полностью
+ повторять функционал существующих. Данные функции записать в прототип.
+
+Пример:
+Array.prototype.customMap = function () {
+  // your implementation here
+}
+*/
+arr = [1,2,3,4,5];
+
+Array.prototype.customMap = function (callback) {
+    if(!Array.isArray(arr)) return console.log('Ошибка');
+
+    const newArray = []
+    for(let index = 0; index < this.length ; index ++){
+        const valueCallback = callback(this[index],index);
```

Comment:
1) better to name `result` or `valueFromCollback`
2) also, callback accept as third argument current array